src/features/CreationForm/ui/CreationForm/CreationForm.tsx

```diff
@@ -0,0 +1,157 @@
+                return yupResolver(thirdStepSchema)(values);
+            }
+            return {};
+        },
+        [step],
+    );
+
+    const onCloseModal = useCallback(() => {
+        setMessage(null);
+    }, [setMessage]);
+
+    const getMessage = useCallback(
+        (result: string | null) => {
+            if (result === null) return null;
+            const title =
+                result === 'success' ? (
+                    <Flex align="center" justify="flex-start">
+                        <b>Форма успешно отправлена</b>
+                    </Flex>
+                ) : (
+                    <Flex align="center" justify="flex-start">
+                        <b>Ошибка</b>
+                    </Flex>
+                );
+            return (
+                <Modal opened centered title={title} onClose={onCloseModal}>
+                    <Message result="success" onClose={onCloseModal} />
+                </Modal>
+            );
+        },
+        [onCloseModal],
+    );
+
+    const onFinish = useCallback(() => {
+        (async () => {
+            await api
+                .post('/content/v1/bootcamp/frontend')
+                .then(() => {
+                    setMessage(getMessage('success'));
+                })
+                .catch(() => {
+                    setMessage(getMessage('error'));
+                });
+        })();
+    }, [setMessage, getMessage]);
+
+    const form = useForm<FormValues>({
+        initialValues: {
+            nickname: '',
+            name: '',
+            surname: '',
+            sex: Gender.Man,
+            about: '',
+            advantages: [],
+            checkbox: [],
+            radio: 0,
+        },
+        validate,
+    });
+
+    const onNext = useCallback(() => {
+        if (form.validate().hasErrors) {
+            return;
+        }
+        if (step === Steps.First) {
+            setStep(Steps.Second);
+            return;
+        }
+        if (step === Steps.Second) {
+            setStep(Steps.Third);
+            return;
+        }
+        if (step === Steps.Third) {
+            onFinish();
+        }
+    }, [form, step, setStep, onFinish]);
+
+    const onBack = useCallback(() => {
+        if (step === Steps.Second) {
+            setStep(Steps.First);
+            return;
+        }
+        if (step === Steps.Third) {
+            setStep(Steps.Second);
+            return;
+        }
+        if (step === Steps.Finish) {
+            setStep(Steps.Second);
+        }
+    }, [step, setStep]);
+
+    const inputs = useMemo(() => {
+        if (step === Steps.First) {
+            return <FirstStep onNext={onNext} onBack={onBack} form={form} />;
+        }
+        if (step === Steps.Second) {
+            return <SecondStep onNext={onNext} onBack={onBack} form={form} />;
+        }
+        if (step === Steps.Third) {
+            return <ThirdStep onNext={onNext} onBack={onBack} form={form} />;
+        }
+        return null;
```

Comment:
Maybe it's better to put this in the markup and not use an extra function?
`{step === Steps.First && <FirstStep onNext={onNext} onBack={onBack} form={form} />}`